openstudiocore/resources/model/OpenStudio.idd

```diff
@@ -1980,6 +1980,39 @@ OS:BuildingStory,
        \units m
        \minimum> 0
 
+OS:BuildingUnit,
+       \min-fields 4
+       \extensible:3
+  A1,  \field Handle
+       \type handle
+       \required-field
+  A2,  \field Name
+       \type alpha
+       \reference BuildingUnitNames
+       \required-field
+  A3,  \field Rendering Color
+       \type object-list
+       \object-list GroupRenderingNames
+  A4,  \field Building Unit Type
+       \type choice
+       \key Residential
+       \key NonResidential
+       \default Residential
+  A5,  \field Building Unit Feature Name
+       \type alpha
+       \begin-extensible
+       \required-field
+  A6,  \field Building Unit Feature Data Type
+       \type choice
+       \key alpha
```

Comment:
@nmerket can you align these keys with the enums [here](https://github.com/NREL/OpenStudio/blob/os_2_0_develop/openstudiocore/src/utilities/data/Variant.hpp#L54):

  ((Boolean)(Boolean)(0))
  ((Double)(Double)(1))
  ((Integer)(Integer)(2))
  ((String)(String)(3))